EIPS/eip-5851.md

```diff
@@ -0,0 +1,257 @@
+
+- Claims: DID Claims are a set of assertions that the DID Holder must fulfill to be verified.
+
+- Holder: The holder is the entity that stores the claim, such as a digital identity provider or a DID registry. The holder is responsible for validating the claim and providing verifiable evidence of the claim.
+
+- Claimer: A claimer is the party making a claim, such as in an identity verification process.
+
+- Issuer: An entity creates a verifiable credential from claims about one or more subjects a holder. Example issuers include governments, corporations, non-profit organizations, trade associations, and individuals.
+
+- Verifier: A verifier in DID is an entity that validates data provided by an issuer of verifiable credentials, determining its accuracy, origin, currency and trustworthiness.
+
+## Motivation
+
+On-chain issuance of verifiable attestations are essential for use-case like:
+
+- Avoiding sybil attacks with one person one vote
+- Participation in certain events with credentials
+- Compliance to the government financial regulations etc.
+
+We are proposing a standard claims structure for DID issuers and verifier entities to create their smart contract in order to provide on chain commitment of the off-chain verification process, and once the given address is associated with the given attestation of the identity verification off-chain, the issuers can then onboard other verifiers (i.e. governance, financial institution, non-profit organization, web3 related cooperation) to define the condition of the ownership of the user in order to reduce the technical barriers and overhead of current implementations.
+
+The motivation behind this proposal is to create a standard for verifier and issuer smart contracts to communicate with each other in a more efficient way. This will reduce the cost of KYC processes, and provide the possibility for on-chain KYC checks.  By creating a standard for communication between verifiers and issuers, it will create an ecosystem in which users can be sure their data is secure and private. This will ultimately lead to more efficient KYC processes and help create a more trustful environment for users. It will also help to ensure that all verifier and issuer smart contracts are up-to-date with the most recent KYC regulations.
+
+## Specification
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Metadata Standard
+
+We will be storing the following structures to store the claims:
+
+### 1. Metadata information
+
+We will store the metadata corresponding ot the each claim requirement, the structure is similar to the [EIP-3475](./eip-3475.md):
+
+```solidity
+    /** Metadata
+    * 
+    * @param title defines the name of the claim field
+    * @param _type is the nature of the data (bool,string,address,bytes,..)
```

Comment:
i think it can be a fixed data type but also a generic type (if solidity will start defining the abstract data type in the next versions) but in any case, this MUST match the type of actual value stored by the Values structure at that stage.

Reply:
The text in the specification (now `kind is the nature of the data (bool,string,address,bytes,..)`) gives no indication of where these values are supposed to come from. If these are supposed to be the Solidity data types, you'll need to actually say that somewhere in your proposal.